wallet/wallet.go

```diff
@@ -2652,7 +2655,11 @@ func (w *Wallet) ImportPrivateKey(scope waddrmgr.KeyScope, wif *btcutil.WIF,
 			return nil
 		}
 
-		return w.Manager.SetBirthday(addrmgrNs, newBirthday)
+		err = w.Manager.SetBirthday(addrmgrNs, newBirthday)
```

Comment:
Agree that this would be nice to have, but seeing how there is currently no testing logic for the wallet and this is release blocking, we should probably address this later? FWIW, `lnd` has no uses of this method.

Reply:
Yeah, I can agree on the test.

But the outlined scenario should be fixed, the new birthday block must be set if it is less than the existing birtdayblock.

Reply:
Fixed.